saleor/graphql/checkout/types.py

```diff
@@ -549,6 +550,10 @@ def with_checkout_info(checkout_info):
     @classmethod
     @traced_resolver
     def resolve_shipping_methods(cls, root: models.Checkout, info):
+        if hasattr(info.context, "sync_event") and info.context.sync_event:
```

Comment:
I see that we use this in many places. Maybe we could create a single decorator that would be used in all resolvers where we need it?

Reply:
great idea. fiuxed